pkg/controller/tidbcluster/tidb_cluster_control.go

```diff
@@ -296,6 +325,75 @@ func (c *defaultTidbClusterControl) updateTidbCluster(tc *v1alpha1.TidbCluster)
 	return err
 }
 
+// NeedToUpdateTiCDCFirst checks if the TiCDC version is greater than or equal to v5.1.0.
+// If so, we need to update TiCDC first before updating other components.
+// See more: https://github.com/pingcap/tidb-operator/issues/4966#issuecomment-1606727165
+func NeedToUpdateTiCDCFirst(ns string, tcName string, deps *controller.Dependencies) (bool, error) {
+	if deps == nil {
+		return false, nil
+	}
+
+	// Check if need to upgrade TiCDC first.
+	upgradeTiCDCFirst := false
+
+	oldSts, err := deps.StatefulSetLister.StatefulSets(ns).Get(controller.TiCDCMemberName(tcName))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return false, fmt.Errorf("updateTidbCluster: failed to get sts %s for cluster %s/%s, error: %s", controller.TiCDCMemberName(tcName), ns, tcName, err)
+	}
+	stsNotExist := apierrors.IsNotFound(err)
+	if !stsNotExist {
```

Comment:
NOTICE: the TiCDC sts wouldn't be created before PD&TiKV is ready, so this will not affect the behavior of TidbCluster creation.